pkg/controller/controller.go

```diff
@@ -233,18 +238,13 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 	if rep.Volume != nil {
 		glog.V(3).Infof("create volume rep: %+v", *rep.Volume)
 	}
-
-	annotations := map[string]string{provisionerIDAnn: p.identity}
-	attributesString, err := json.Marshal(rep.Volume.Attributes)
-	if err != nil {
-		glog.V(2).Infof("fail parsing volume attributes: %+v", rep.Volume.Attributes)
-	} else {
-		annotations[volumeAttributesAnnotation] = string(attributesString)
+	volumeAttributes := map[string]string{provisionerIDKey: p.identity}
```

Comment:
This sends `storage.kubernetes.io/csiProvisionerIdentity` to CSI driver on all subsequent calls. I guess drivers may not like this and may reject requests with unknown attributes.

Reply:
I see your point, I will bring it up on CSI WG meeting, same as the reason for provisioner identity checking.